diff --git a/web/dashboard.jsp b/web/dashboard.jsp
index 134b7f3..7a09318 100644
--- a/web/dashboard.jsp
+++ b/web/dashboard.jsp
@@ -180,7 +180,7 @@
             var values = getDataValues(data);
             // Convert timestamps to string; HighCharts already defines a nice formatting one.
             for (i = 0; i < timeStamps.length; i++) {
-                timeStampStr.push([new Date(timeStamps[i]), values[0][i]]);
+                timeStampStr.push([timeStamps[i], values[0][i]]);
             }
 
             // Custom this to set theme, see: http://www.highcharts.com/docs/chart-design-and-style/design-and-style
diff --git a/web/scripts/dashboard.js b/web/scripts/dashboard.js
index d5f15a2..db3972c 100644
--- a/web/scripts/dashboard.js
+++ b/web/scripts/dashboard.js
@@ -139,7 +139,7 @@ function fetchData(json) {
     for (var i = 0; i < values.length; i++) {
         var arr = [];
         for (var j = 0; j < timeStamps.length; j++) {
-            arr.push([new Date(timeStamps[j]), values[i][j]]);
+            arr.push([timeStamps[j], values[i][j]]);
             console.log("Pushed: " + values[i][j]);
         }
         timeStampStr.push(arr);
